src/GafferSceneUI/TransformTool.cpp

```diff
@@ -157,6 +158,17 @@ bool updateSelection( const SceneAlgo::History *history, TransformTool::Selectio
 	if( selection.transformPlug )
 	{
 		selection.transformPlug = selection.transformPlug->source<TransformPlug>();
+
+		if( auto *spreadsheet = runTimeCast<Spreadsheet>( selection.transformPlug->node() ) )
+		{
+			if( spreadsheet->outPlug()->isAncestorOf( selection.transformPlug.get() ) )
+			{
+				selection.transformPlug = static_cast<TransformPlug *>(
+					spreadsheet->activeInPlug( selection.transformPlug.get() )
+				);
+			}
+		}
+
```

Comment:
Is it intended that this then edits the default row, if no other row in the spread sheet matches? Feels like that could cause unwanted side effects?

Reply:
Good point. adf3303 and 5cacddc.

Reply:
With the following (sheet with a single row) I can't edit with the translate tool any more:

```
import Gaffer
import GafferScene
import IECore
import imath

Gaffer.Metadata.registerValue( parent, "serialiser:milestoneVersion", 0, persistent=False )
Gaffer.Metadata.registerValue( parent, "serialiser:majorVersion", 55, persistent=False )
Gaffer.Metadata.registerValue( parent, "serialiser:minorVersion", 1, persistent=False )
Gaffer.Metadata.registerValue( parent, "serialiser:patchVersion", 0, persistent=False )

__children = {}

__children["Plane"] = GafferScene.Plane( "Plane" )
parent.addChild( __children["Plane"] )
__children["Plane"].addChild( Gaffer.V2fPlug( "__uiPosition", defaultValue = imath.V2f( 0, 0 ), flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic, ) )
__children["Transform"] = GafferScene.Transform( "Transform" )
parent.addChild( __children["Transform"] )
__children["Transform"].addChild( Gaffer.V2fPlug( "__uiPosition", defaultValue = imath.V2f( 0, 0 ), flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic, ) )
__children["PathFilter"] = GafferScene.PathFilter( "PathFilter" )
parent.addChild( __children["PathFilter"] )
__children["PathFilter"].addChild( Gaffer.V2fPlug( "__uiPosition", defaultValue = imath.V2f( 0, 0 ), flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic, ) )
__children["Spreadsheet"] = Gaffer.Spreadsheet( "Spreadsheet" )
parent.addChild( __children["Spreadsheet"] )
__children["Spreadsheet"]["rows"].addColumn( Gaffer.V3fPlug( "translate", defaultValue = imath.V3f( 0, 0, 0 ), ) )
__children["Spreadsheet"]["rows"].addRows( 1 )
__children["Spreadsheet"].addChild( Gaffer.V2fPlug( "__uiPosition", defaultValue = imath.V2f( 0, 0 ), flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic, ) )
__children["Plane"]["__uiPosition"].setValue( imath.V2f( 2.04999971, 9.75000095 ) )
__children["Transform"]["in"].setInput( __children["Plane"]["out"] )
__children["Transform"]["filter"].setInput( __children["PathFilter"]["out"] )
__children["Transform"]["transform"]["translate"].setInput( __children["Spreadsheet"]["out"]["translate"] )
__children["Transform"]["transform"]["translate"]["x"].setInput( __children["Spreadsheet"]["out"]["translate"]["x"] )
__children["Transform"]["transform"]["translate"]["y"].setInput( __children["Spreadsheet"]["out"]["translate"]["y"] )
__children["Transform"]["transform"]["translate"]["z"].setInput( __children["Spreadsheet"]["out"]["translate"]["z"] )
__children["Transform"]["__uiPosition"].setValue( imath.V2f( 2.04999971, -1.24999988 ) )
__children["PathFilter"]["paths"].setValue( IECore.StringVectorData( [ '/.../*' ] ) )
__children["PathFilter"]["__uiPosition"].setValue( imath.V2f( 15.8500004, 5.13203144 ) )
Gaffer.Metadata.registerValue( __children["Spreadsheet"], 'nodeGadget:type', 'GafferUI::StandardNodeGadget' )
__children["Spreadsheet"]["selector"].setValue( '${scene:path}' )
__children["Spreadsheet"]["rows"]["row1"]["name"].setValue( '/plane' )
__children["Spreadsheet"]["rows"]["row1"]["cells"]["translate"]["value"].setValue( imath.V3f( 1, 1, 0 ) )
Gaffer.Metadata.registerValue( __children["Spreadsheet"]["rows"]["default"]["cells"]["translate"]["value"], 'nodule:color', imath.Color3f( 0.469999999, 0.180999994, 0.180999994 ) )
Gaffer.Metadata.registerValue( __children["Spreadsheet"]["rows"]["default"]["cells"]["translate"]["value"]["x"], 'nodule:color', imath.Color3f( 0.246700004, 0.376199991, 0.469999999 ) )
Gaffer.Metadata.registerValue( __children["Spreadsheet"]["rows"]["default"]["cells"]["translate"]["value"]["y"], 'nodule:color', imath.Color3f( 0.246700004, 0.376199991, 0.469999999 ) )
Gaffer.Metadata.registerValue( __children["Spreadsheet"]["rows"]["default"]["cells"]["translate"]["value"]["z"], 'nodule:color', imath.Color3f( 0.246700004, 0.376199991, 0.469999999 ) )
__children["Spreadsheet"]["__uiPosition"].setValue( imath.V2f( -16.1491661, -1.2495836 ) )


del __children
```

Reply:
You need to add the entire transform plug to the spreadsheet as one, not just promote translate on its own. That's what the "Create Spreadsheet (Transform)" and "Add to Spreadsheet (Transform)" menu items are for, since there isn't a right click menu for the entire transform plug.

Reply:
You need to add the entire transform plug to the spreadsheet as one, not just promote translate on its own. That's what the "Create Spreadsheet (Transform)" and "Add to Spreadsheet (Transform)" menu items are for, since there isn't a right click menu for the entire transform plug.